onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -818,6 +827,58 @@ void addGlobalMethods(py::module& m, Environment& env) {
           throw std::runtime_error("Error when creating and registering allocator: " + st.ErrorMessage());
         }
       });
+#ifdef USE_CUDA
+  m.def("numpy_array_to_cuda", [](const py::array& cpu_src, const OrtDevice& ort_device) -> py::array {
```

Comment:
Not understanding why we need these global methods ? To get numpy conetns to CUDA, can't we just create an OrtValue from numpy specifying OrtDevice as CUDA ? Why expose something that already exists ?

Reply:
SparseTensor is created on top of the user buffers that are numpy arrays, not OrtValues. So I needed something on cuda and test moving SparseTensor to cpu and back to cuda.

Reply:
Is it not sufficient to create an OrtValue containing a PySparseTensor on CPU and use to_cuda via that? 

i.e. do we need this low level helper or can we keep it simple and just assume all data for the SparseTensor (vs. an individual array) is one the same device, and if we do a copy to a different device it's via the OrtValue (that contains the PySparseTensor).

Reply:
If you want a SparseTensor on cuda, call SparseTensor.to_cuda, not OrtValue. OrtValue is a container to feed/fetch data to run the model.

Reply:
You mean create SparseTensor on cpu, use to_cuda and then create OrtValue from that? yes, it is possible.

Reply:
I removed those methods. Now, if you get SparseTensor on GPU, you will need to copy data yourself.